### PR TITLE
Add javadoc for ConnectorSplit#isRemotelyAccessible

### PR DESCRIPTION
**Pull request by raunaqmorarka**
## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

File: core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java [hunk 1]
```diff
@@ -20,6 +20,13 @@
 
 public interface ConnectorSplit
 {
+    /**
+     * Returns whether a ConnectorSplit can be scheduled on any node.
```
Comment (nineinchnick) [inline core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java:24]:
```suggestion
     * Returns true whether this ConnectorSplit can be scheduled on any node.
```

File: core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java [hunk 2]
```diff
@@ -20,6 +20,13 @@
...
+     * <p>
+     * When true, the addresses returned by {@link #getAddresses()} are treated by the scheduler
+     * as hints, which it may not follow strictly.
+     * When false, the split will always be scheduled on one of the addresses returned by {@link #getAddresses()}.
```
Comment (nineinchnick) [inline core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java:28]: Both these cases are the same, that the split will be scheduled on one of the nodes returned by `getAddresses()`.

Comment (raunaqmorarka) [reply core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java]: I've updated the comment now

File: core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java [hunk 3]
```diff
@@ -20,6 +20,13 @@
 
 public interface ConnectorSplit
 {
+    /**
+     * Returns true when this ConnectorSplit can be scheduled on any node.
+     * <p>
+     * When true, the split may be scheduled on any node. The addresses returned by {@link #getAddresses()}
```
Comment (Praveen2112) [inline core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java:26]: Not sure but `the split may be scheduled on any node` is redundant right ?

**Review by nineinchnick — APPROVED** [review]

File: core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java (hunk 3 above)
Comment (raunaqmorarka) [reply core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorSplit.java]: Yes, made it a bit more succinct now

**Review by martint — APPROVED** [review]

**Pull request merged**